whois-update/src/main/java/net/ripe/db/whois/update/handler/transform/AsSetSourceAttributeTransformer.java

```diff
@@ -0,0 +1,71 @@
+package net.ripe.db.whois.update.handler.transform;
+
+import net.ripe.db.whois.common.dao.RpslObjectDao;
+import net.ripe.db.whois.common.domain.CIString;
+import net.ripe.db.whois.common.rpsl.AttributeType;
+import net.ripe.db.whois.common.rpsl.ObjectType;
+import net.ripe.db.whois.common.rpsl.RpslAttribute;
+import net.ripe.db.whois.common.rpsl.RpslObject;
+import net.ripe.db.whois.common.rpsl.RpslObjectBuilder;
+import net.ripe.db.whois.update.domain.Action;
+import net.ripe.db.whois.update.domain.Update;
+import net.ripe.db.whois.update.domain.UpdateContext;
+import net.ripe.db.whois.update.domain.UpdateMessages;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+@Component
+public class AsSetSourceAttributeTransformer implements Transformer{
+
+    private final RpslObjectDao rpslObjectDao;
+
+    @Value("${whois.nonauth.source}")
+    private String nonAuthSource;
+
+    public AsSetSourceAttributeTransformer(final RpslObjectDao rpslObjectDao){
+        this.rpslObjectDao = rpslObjectDao;
+    }
+
+    @Override
+    public RpslObject transform(final RpslObject rpslObject, final Update update,
+                                final UpdateContext updateContext,
+                                final Action action) {
+        if (Action.DELETE.equals(action) || !ObjectType.AS_SET.equals(rpslObject.getType())) {
+            return rpslObject;
+        }
+        final String asSetKey = rpslObject.getKey().toString();
+
+        boolean flatAsSet = !asSetKey.contains(":");
+        final CIString asSetSource = rpslObject.getValueForAttribute(AttributeType.SOURCE);
+
+        if (flatAsSet){
+            if (asSetSource != null) {
+                updateContext.addMessage(update, UpdateMessages.flatModelNotAllowSourceModifications(asSetSource.toString()));
```

Comment:
asSetSource can neve be null it is mandatory. Also you are using getValueForAttribute which will throw exception

Reply:
I am changing this